### ISSUE TITLE
TrimFields and CurrentRecord

### ISSUE THREAD

**Issue by JimF42** [JoshClose/CsvHelper#443]
I am using TryGetField to get my data normally, but when it fails I am using CurrentRecord to report back what offending data was actually in the field. 

TrimFields = true appears to be honored with the [Try]GetField methods, but when I fall back to CurrentRecord, it is not honored and I get back the untrimmed field.  Is this by design? Of course, I can workaround this if it is by design, but I wanted to report this if it was a bug.

**Comment by roadrunnerlenny** [comment]
I have the same issue with `CsvReader.CurrentRecord` and untrimmed fields.
I set `TrimFields = true` in the `CsvReader.Configuration`. When I access an already read line via `CsvReader.CurrentRecords`, the fields are untrimmed.
If I access each element of the line via the `CsvReader.GetField(int index)` - method, I get a trimmed value.
As I am interested in the whole line as an array of string, I found this is a valid workaround:
```
string[] line = new string[CsvReader.CurrentRecord.Length];
for (int idx = 0; idx < CsvReader.CurrentRecord.Length; idx++)
  line[idx] = CsvReader.GetField(idx);
```

Same thing for the header. I set `CsvReader.Configuration.TrimHeaders = true;`, but when I access the FieldHeaders via  CsvReader.FieldHeaders, I get the untrimmed values. This worked for me then:
`CsvReader.FieldHeaders.Select(header => header.Trim()).ToArray();`
